src/components/smartSearch/SmartSearchDialog/QueryOverview.tsx

```diff
@@ -17,91 +17,127 @@ import { AnyFilterConfig, CallHistoryFilterConfig, CampaignParticipationConfig,
     PersonDataFilterConfig, PersonTagsFilterConfig, RandomFilterConfig, SmartSearchFilterWithId,
     SurveyResponseFilterConfig, SurveySubmissionFilterConfig, UserFilterConfig } from 'types/smartSearch';
 
+const FIRST_FILTER = 'all';
+
 interface QueryOverviewProps {
     filters: SmartSearchFilterWithId<AnyFilterConfig>[];
     onCloseDialog: () => void;
+    onAddNewFilter: (type: FILTER_TYPE) => void;
     onSaveQuery: () => void;
     onOpenFilterGallery: () => void;
     onEditFilter: (filter: SmartSearchFilterWithId) => void;
     onDeleteFilter: (filter: SmartSearchFilterWithId) => void;
 }
 
 const QueryOverview = (
-    { filters, onCloseDialog, onSaveQuery, onOpenFilterGallery, onEditFilter, onDeleteFilter }:QueryOverviewProps,
+    { filters, onCloseDialog, onAddNewFilter, onSaveQuery, onOpenFilterGallery, onEditFilter, onDeleteFilter }:QueryOverviewProps,
 ): JSX.Element => {
-    const [hovered, setHovered] = useState<number | null>(null);
+    const [hovered, setHovered] = useState<number | null | string>(null);
+    const firstFilter = filters[0];
+    const startWithEveryone =  firstFilter?.type === FILTER_TYPE.ALL && firstFilter.op === 'add';
```

Comment:
What happens if the first filter is `FITER_TYPE.ALL` and `sub`? Could that ever happen? If it can't happen the condition is unnecessary (and I think that's the case, since the normalization will have removed such a filter).